### PR TITLE
Add 'How to read the Godot API' docs.

### PR DESCRIPTION
**Pull request by willnationsdev**
An introduction to the format of the Godot API class reference pages.

This is a partial rewrite of portions of #2958 

cc @NathanLovato 

Notes:

1. I put it in the getting started section since that's what #4035 seemed to mention. Let me know if it should be moved.
2. For some reason, many of the images I take with Windows Snippet end up blurry even though they look fine in the app. Also tried manually screenshotting and cropping in Paint, but that also lead to a blurry result on the webpage. Any suggestions?

### CODE REVIEW

**Comment by NathanLovato** [comment]
The idea is not to provide a reference of the reference but a guide on how to learn new things in Godot using the docs. See  #4169 

Your page is describing the reference page, which I don't find too useful for new users, whose problems are more like "how do I move objects around the screen?", "what can I do with this node?". Multiple parts of the reference pages like tutorials, descriptions, brief description are self-explanatory. I would mention in one or two paragraphs what you can find on a reference page, explain the inheritance, (the current class has all properties and methods from its parents), and most importantly how to search using it.

In short, you don't want to provide a reference describing the reference, but an actionable guide on using and searching the documentation in general, and tips to find the information you're looking for.

I'd also recommend writing what the user will learn at the top of every page you write. Like "This page explains ..." or "In this guide, you will learn...". Doing so

1. Tells the user whether that's the page they're looking for in 10 seconds.
2. Helps you frame your guide in terms of the techniques you teach.

**Comment by NathanLovato** [comment]
Reviewing the page again, I think it's good to have. I'd move it to the scripting section or another one. Not sure which is best, but scripting seems close enough.

**Review by Calinou — APPROVED** [review]
Should be good to merge after rebasing and moving the page to the Scripting section.

**Comment by akien-mga** [comment]
> Should be good to merge after rebasing and moving the page to the Scripting section.

Bump :)

**Comment by willnationsdev** [comment]
1. Rebased
2. Moved code to Scripting section.
3. Removed the empty spaces beneath each image (ended up looking really weird in the final version).
4. Changed a reference to Callable to be a link.

**Comment by willnationsdev** [comment]
bump. Should be ready to merge, afaik.

**Comment by NathanLovato** [comment]
Thanks for your work!
